tensorboard/webapp/webapp_data_source/tb_http_client.ts

```diff
@@ -84,7 +84,7 @@ export class TBHttpClient implements TBHttpClientInterface {
     path: string,
     // Angular's HttpClient is typed exactly this way.
     body: any | null,
-    options: PostOptions = {}
+    options: PostOptions | undefined = {}
```

Comment:
What does it mean for this to have both undefined and a default value of {}? Does the default value mean it's still an optional property? But there are still some cases where you want to explicitly set it to undefined?

Reply:
Unfortunately, TypeScript does not seem to understand that `options` cannot be undefined when having default value like this. `undefined` was introduce TypeScript compiler and that's all. 

No, there isn't a case where we would like `options` to ever be `undefined` but we still guard against that for now.